grpc4bmi/bmi_client_singularity.py

```diff
@@ -29,35 +29,92 @@ class BmiClientSingularity(BmiClient):
     The client picks a random port and expects the container to run the server on that port.
     The port is passed to the container using the BMI_PORT environment variable.
 
-    >>> from grpc4bmi.bmi_client_singularity import BmiClientSingularity
-    >>> image = 'docker://ewatercycle/wflow-grpc4bmi:latest'
-    >>> client = BmiClientSingularity(image, input_dir='wflow_rhine_sbm', output_dir='wflow_output')
-    >>> client.initialize('wflow_rhine_sbm/wflow_sbm.ini')
-    >>> client.update_until(client.get_end_time())
-    >>> del client
-
     Args:
         image: Singularity image. For Docker Hub image use `docker://*`.
-        input_dir (str): Directory for input files of model
-        output_dir (str): Directory for input files of model
+        input_dirs (Iterable[str]): Input directories on host computer.
+
+            All of them will be mounted read-only inside Singularity container on same path as outside container.
+
+        work_dir (Optional[str]): Working directory for model.
```

Comment:
Having rw and ro binds of same dir is indeed tricky so I now raise an exception when work_dir is the same as one of input dirs. 

There are usecases ([example 1 and 2](https://github.com/eWaterCycle/grpc4bmi/blob/d55b2500b9d3ecf3978d8d47986d54a286f848e8/grpc4bmi/bmi_client_singularity.py#L44-L68)) where work dir can be optional. I think we should support those use cases.

Reply:
As I understand it not specifying a work_dir puts all the output and temp files in the current working directory of the user. I suspect this will be a more used by users often since they will not grasp the consequences. The user will then be confused when running two models at the same time suddenly messes up the model.

I thrfore think not specifying a work_dir should either be an error, or lead to the work_dir remaining only inside the container.

Reply:
In singularity:
Use `--contain` to not have any surprise mounts. And make work_dir required. Document how to create a temp work folder.

Reply:
done